azure-devops/azext_devops/dev/pipelines/pipeline_create.py

```diff
@@ -71,6 +72,9 @@ def pipeline_create(name, description=None, repository=None, branch=None, yml_pa
     :type service_connection: str
     :param queue_id: Id of the queue in the available agent pools. Will be auto detected if not specified.
     :type queue_id: str
+    :param skip_first_run: Specify this flag to prevent the first run being triggered by the command.
+    Command will return a pipeline if run is skipped else it will output a pipeline run.
```

Comment:
`Command will return a pipeline if run is skipped else it will output a pipeline run.`
Is this something we are ok with? same command returning different type of response (based on different inputs)

Reply:
So I see three concerns: 
1. Deterministic output
2. Command should be atomic and never trigger a run
3. If we have an option to trigger a run it should be opt-in rather than opt-out

For 1, we have had a thread with Travis Prescott on email which says - It is ok for a command to call different APIs and return different outputs. "From the perspective of the output, the output is still deterministic based on what the user specifies, so that is not a concern."
Is it not same as ```az devops service-endpoint show```?

For 2 and 3, the selected default behavior is due to two reasons:
1. Match UI behaviour since UI does not even give an option to opt-out, same is by default in the command.
2. Yaml issues are only found during the run so an invalid pipeline creation is caught early by triggering a run (which in most cases should be harmless if you are setting up a CI). It will catch issues like resource permissions etc.

@geverghe  To take a call. 

cc: @gauravsaralMs @ishitam8 @snankani

Reply:
@atbagga has captured the reasoning quite accurately and I concur with the same. Few additional clarifications on point 2,3
1. I had a word with @vijayma, and the 80% scenario is that the experience for creating a pipeline is to have the pipeline built and the first run kick started. Without it, it's just a commit of another file. Further, the objective is to ensure at least one run to ensure things are in order.  
2. For the experience that is visualized, if we make it atomic, then the cognitive load to achieve the ideal experience increases from CLI. Considering we have a command working pretty efficiently at this point, I feel that we should be good tweaking the experience to include the skip part.